k8s/utils/symphony-api.go

```diff
@@ -238,3 +239,28 @@ func CreateSymphonyDeployment(ctx context.Context, instance solution_v1.Instance
 
 	return ret, err
 }
+
+func NeedWatchInstance(instance solution_v1.Instance) bool {
+	var interval time.Duration = 30
+	if instance.Spec.ReconciliationPolicy != nil && instance.Spec.ReconciliationPolicy.Interval != nil {
+		parsedInterval, err := time.ParseDuration(*instance.Spec.ReconciliationPolicy.Interval)
+		if err != nil {
+			parsedInterval = 30
+		}
+		interval = parsedInterval
+	}
+
+	if instance.Spec.ReconciliationPolicy != nil && instance.Spec.ReconciliationPolicy.State.IsInActive() || interval == 0 {
+		return false
+	}
+
+	return true
+}
+
+func ReplaceLastSeperator(name string, seperatorBefore string, seperatorAfter string) string {
+	i := strings.LastIndex(name, seperatorBefore)
+	if i == -1 {
+		return name
+	}
+	return name[:i] + seperatorAfter + name[i+1:]
```

Comment:
If we'll support "-" in names again, we need to check if "-" is the last character

Reply:
Yes, it is right now. This function is to split string based on last "-". And form it to ":" as seperator.